sync-tool/UpdateFromPerlSrc.php

```diff
@@ -0,0 +1,189 @@
+        $this->tokenizer->addBracket('[', ']');
+        $this->tokenizer->addDelimiter(';');
+    }
+
+    private function readXxxPm($content) {
+        $tokens = $this->tokenizer->tokenize($content);
+        $renderer = new \Yaoi\String\Lexer\Renderer();
+        $renderer->
+            setBindKey('-~z', 'z~-')
+            ->strip('#')
+            ->keepBoundaries('[');
+
+        $expression = $renderer->getExpression($tokens);
+        $statement = $expression->getStatement();
+        /** @var \Yaoi\String\Lexer\Parsed[] $binds */
+        $binds = $expression->getBinds();
+
+
+        $parser = new \Yaoi\String\Parser($statement);
+        $block = (string)$parser->inner('$Text::Unidecode::Char[', ']');
+        if (!$block) {
+            throw new \Exception('Block not found');
+        }
+        $block = $renderer->getExpression($binds[$block])->getStatement();
+
+        $php = <<<PHP
+<?php
+\$UTF8_TO_ASCII[$block] = array(
+
+PHP;
+
+        $itemsBind = (string)$parser->inner('[', ']');
+        $phpFilePath = __DIR__ . '/../src/Behat/Transliterator/data/' . substr($block, 1) . '.php';
+
+        if (!$itemsBind) {
+            $this->response->addContent('Empty char table for block ' . $block);
+            if (file_exists($phpFilePath)) {
+                if (unlink($phpFilePath)) {
+                    $this->response->success('Deleted');
+                }
+                else {
+                    $this->response->error('Failed to delete');
+                }
+            } else {
+                $this->response->success('No PHP file, skipped');
+            }
+            //echo $statement, PHP_EOL;
+            return;
+        }
+        $items = $binds[$itemsBind];
+        $itemsExpression = $renderer->getExpression($items);
+        $itemsStatement = $itemsExpression->getStatement();
+        $itemsBinds = $itemsExpression->getBinds();
+
+        $itemsStatement = explode(',', $itemsStatement);
+        $index = 0;
+        $nonQuestionBoxFound = false;
+        foreach ($itemsStatement as $item) {
+            $item = trim($item);
+            if (!$item) {
+                break;
+            }
+
+            if ($index >= 16) {
+                $php = trim($php);
+                $php .= PHP_EOL;
+                $index = 0;
+            }
+            ++$index;
+
+            $value = $item;
+
+            if (isset($itemsBinds[$item])) {
+                /** @var \Yaoi\String\Lexer\Token $token */
+                $token = $itemsBinds[$item];
+                $value = $token->unEscapedContent;
+            }
+
+
+            $value = new StringValue($value);
+            if ($value->starts('\x')) {
+                $php .= '"' . $value . '", ';
+                $nonQuestionBoxFound = true;
+            } else {
+                // TODO check if this hack should be removed for chinese letters
+                if ($value->value === '[?] ') {
+                    $value->value = '[?]';
+                }
```

Comment:
In chinese tables many symbols are translated to `[?] ` (with space), in PHP tables they are trimmed. Can you confirm space is redundant or needed?